### PR TITLE
WIP: Pluralise Transaction fields

### PR DESCRIPTION
**Pull request by tcharding**
This has been mention a few times now, I went  to quickly hack it up but am getting unexplainable test fails for something that should not effect tests. Is this a bug in our code or in my patch?

Anyways, thought I'd throw it up as draft.

Fix: #822

### CODE REVIEW

**Comment by tcharding** [comment]
cc @Kixunil

**Review by Kixunil — COMMENTED** [review]
Cool! I thought it'd be larger :)

And that test failure wins the weirdest failure of the year.

**Comment by Kixunil** [comment]
Oh and can we wait for my big PRs to merge?

**Comment by tcharding** [comment]
> Oh and can we wait for my big PRs to merge?

I couldn't resist doing this since you kept mentioning it and after our recent rebase tit-tats :) I want to see it done, but I was f**king with you too :)

File: bitcoin/examples/sighash.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ use hex_lit::hex;
 /// * `value` - ref tx output value in sats
 fn compute_sighash_p2wpkh(raw_tx: &[u8], inp_idx: usize, value: u64) {
     let tx: Transaction = consensus::deserialize(raw_tx).unwrap();
-    let inp = &tx.input[inp_idx];
```
Comment (yancyribbens) [inline bitcoin/examples/sighash.rs:26]: `inp` seems weird to me. why not just `i` or `input`

**Comment by yancyribbens** [comment]
It's PR's like this that make me wonder why was this singular to begin with.

**Comment by apoelstra** [comment]
> It's PR's like this that make me wonder why was this singular to begin with.

Because `input[i]` reads as "input i" which is more natural and grammatical than `inputs i".

But as always, I don't care about stylistic things and if people want to change stuff like this, and don't get opposition, they can go ahead.

**Comment by dpc** [comment]
> Because `input[i]` reads as "input i" which is more natural and grammatical than `inputs i".

BTW. To go on a tangent about more general principle: Trying to bend the code to look and behave somewhat like a natural language prose is a common source of weirdness and confusion. My favorite example being all the test assertion libraries popping in all programming languages all the time, like e.g. [npm's `should`](https://www.npmjs.com/package/should) with little nightmares like `should(user).have.property('name', 'tj');`

The problem with it is that the rest of code is not written in a natural language like way, and mixing these "modes" creates uncertainty and confusion.

When read in the cold mechanical meaning of the programming language `tx.input[3]` reads as "transaction always have an input and that input is a container". Importantly the name "input" describes the field/value itself, not the values it contains - containers are no different than single values in this respect.

**Comment by Kixunil** [comment]
@dpc interesting example! Though I think it can work when the whole language does it. Like Swift `writeBytes(bytes, toFile: file)`. (I'm kinda jealous of Swift having named parameters.)

And speaking of making code look like other - in Rust generally plurals are used for containers so it's more consistent. And also I never though of it being as "input container" because when I'm discussing with other people it's either "one input" which is not container or "all inputs" which is but it's plural.

**Comment by tcharding** [comment]
We aren't doing this anymore. We added pluralised getters on the `Transaction` when we moved it to `primitives`.

**Pull request closed without merging**